crates/core/app/tests/mock_consensus.rs

```diff
@@ -80,54 +87,90 @@ async fn mock_consensus_can_send_a_spend_action() -> anyhow::Result<()> {
-            .ok_or_else(|| anyhow!("commitment is not witnessed"))?
-            .position();
-        let spend = SpendPlan::new(&mut rng, note.clone(), position);
+            .witness(commitment.clone())
+            .ok_or_else(|| anyhow!("commitment is not witnessed"))?;
+        let note = note.clone();
+        let position = proof.position();
+        SpendPlan::new(&mut OsRng, note, position).into()
+    };
+
+    // Build a transaction spending this note.
+    let tx: Transaction = {
+        let chain_id = TestNode::<()>::CHAIN_ID.to_string();
+        let transaction_parameters = TransactionParameters {
+            expiry_height: 0,
+            fee: Fee::default(),
+            chain_id,
+        };
+        let detection_data = Some(DetectionDataPlan {
+            clue_plans: vec![CluePlan::new(&mut OsRng, *test_keys::ADDRESS_0, 0)],
+        });
         let plan = TransactionPlan {
-            actions: vec![spend.into()],
-            ..Default::default()
+            actions: vec![
+                spend,
+                OutputPlan::new(&mut OsRng, note.value(), *test_keys::ADDRESS_1).into(),
+            ],
+            transaction_parameters,
+            detection_data,
+            memo: Some(MemoPlan::new(
+                &mut OsRng,
+                MemoPlaintext::blank_memo(*test_keys::ADDRESS_1),
+            )?),
+        };
+        let witness = WitnessData {
+            anchor: client.sct.root(),
+            state_commitment_proofs: plan
+                .spend_plans()
+                .map(|spend| {
+                    (
+                        spend.note.commit(),
+                        client.sct.witness(spend.note.commit()).unwrap(),
+                    )
+                })
+                .collect(),
         };
-        let witness = plan.witness_data(&client.sct)?;
-        let auth = plan.authorize(rand_core::OsRng, spend_key)?;
+        let auth = plan.authorize(OsRng, spend_key)?;
         plan.build_concurrent(viewing_key, &witness, &auth).await?
     };
 
     // Execute the transaction, and sync another mock client up to the latest snapshot.
     test_node
         .block()
-        .with_data(vec![tx.encode_to_vec()]) // TODO(kate): add a `with_tx` extension method
+        .with_data(vec![tx.encode_to_vec()])
         .execute()
         .await?;
 
     // Sync to the latest storage snapshot once more.
-    let client = MockClient::new(test_keys::FULL_VIEWING_KEY.clone())
+    let client_after_spend = MockClient::new(viewing_key.deref().clone())
         .with_sync_to_storage(&storage)
-        .await?;
+        .await?
+        .tap(|c| info!(client.notes = %c.notes.len(), "mock client synced to test storage"));
 
-    client.notes.get(&commitment).unwrap();
+    // Show that we performed the spend as expected.
+    assert_eq!(
+        client_after_spend.notes.len(),
+        client.notes.len() + 1,
+        "a new note should exist after performing the spend",
+    );
```

Comment:
cc: @hdevalence

in our [previous conversation](https://github.com/penumbra-zone/penumbra/issues/3876#issuecomment-1961951033) in #3876, we came to the decision that we should make some assertions here to show that the spend was performed properly.

i'm not entirely familiar with the underlying model w.r.t `Note`s. i have _an_ assertion, above, but is there more that we should be asserting here?

Reply:
Added in 54c0b35d8 :

- we create pre- and post- snapshots of the chain, and check that the nullifiers were correctly revealed
- we check the mock client for the exact output note commitment we're expecting

Reply:
you resolved this in https://github.com/penumbra-zone/penumbra/pull/3891/commits/54c0b35d82802c08b3a48cc2a1008c69951eaa9b.